### PR TITLE
Refactor build scripts

### PR DESCRIPTION
**Pull request by LukeMathWalker**
Feature-flag all testing-related tasks.
Remove `dokka` in order to have no test-specific third-party dependency.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Comment by Velfi** [comment]
@eduardomourar We've decided to remove `dokka` due to the maintenance burden it introduced and the fact that none of the core maintainers currently make use of it.

**Review by Velfi — APPROVED** [review]

File: codegen-server/python/build.gradle.kts [hunk 1]
```diff
@@ -56,15 +52,27 @@ val sourcesJar by tasks.creating(Jar::class) {
...
-    useJUnitPlatform()
-    testLogging {
-        events("passed", "skipped", "failed")
-        exceptionFormat = TestExceptionFormat.FULL
-        showCauses = true
-        showExceptions = true
-        showStackTraces = true
-        showStandardStreams = true
+val isTestingEnabled: String by project
+if (isTestingEnabled.toBoolean()) {
```
Comment (jjant) [inline codegen-server/python/build.gradle.kts:56]: 🤔  Doesn't `val isTestingEnabled: Boolean by project` work?

Comment (jjant) [inline codegen-server/python/build.gradle.kts:56]: Doesn't `val isTestingEnabled: Boolean by project` work?

File: aws/sdk-codegen/build.gradle.kts [hunk 2]
```diff
@@ -78,29 +70,44 @@ val sourcesJar by tasks.creating(Jar::class) {
...
-        csv.required.set(false)
-        html.outputLocation.set(file("$buildDir/reports/jacoco"))
+    tasks.compileTestKotlin {
+        kotlinOptions.jvmTarget = "1.8"
     }
-}
 
-// Always run the jacoco test report after testing.
-tasks["test"].finalizedBy(tasks["jacocoTestReport"])
+    tasks.test {
```
Comment (crisidev) [inline aws/sdk-codegen/build.gradle.kts:87]: I think it worth to add an issue to reduce the code duplication and add the testing logic to `buildSrc`.

**Review by crisidev — APPROVED** [review]

File: aws/sdk-codegen/build.gradle.kts (hunk 2 above)
Comment (crisidev) [reply aws/sdk-codegen/build.gradle.kts:87]: https://github.com/awslabs/smithy-rs/issues/2118

File: codegen-server/python/build.gradle.kts (hunk 1 above)
Comment (LukeMathWalker) [reply codegen-server/python/build.gradle.kts:56]: Nope 😭

**Pull request merged**